contract.py

```diff
@@ -80,20 +78,25 @@ def configure(self, period: arc4.UInt64) -> None:
     # - total and funding are set to arguments
     ##############################################
     @arc4.abimethod
-    def fill(self, total: arc4.UInt64, funding: arc4.UInt64) -> None:
-        self.enforce_step(UInt64(2)) # Ready
-        self.require_funder()
-        # JM: It is actually not necessary to have `total` as an argument that you check like this. Instead, you can inspect what the payment actually is and then that's what "total" is
-        self.require_payment(self.funder, total.native)
-        assert total > 0, "payment is greater than zero"
-        self.total = total.native
+    def fill(self, funding: arc4.UInt64) -> None:
+        ##########################################
+        assert self.owner != Global.zero_address, "owner initialized"
+        assert self.funding == 0, "funding not initialized"
+        ##########################################
+        assert Txn.sender == Global.creator_address, "must be creator" 
+        ##########################################
+        payment_amount = self.require_payment(Global.creator_address)
```

Comment:
Not that important, but it is okay if the payment doesn't come from the creator, provided the creator is the one that sends the `appl`. This COULD make it easy for the Foundation to have different wallets for different purposes.